app/plugin.go

```diff
@@ -557,3 +637,59 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+func (a *App) signaturesFromPathToReader(plugin *pluginSignaturePaths) (signatures []io.ReadSeeker, closeFunc func(), appErr *model.AppError) {
+	signatures = []io.ReadSeeker{}
+	for _, signaturePath := range plugin.signaturePaths {
+		sigReader, appErr := a.FileReader(signaturePath)
+		if appErr != nil {
+			mlog.Debug("Can't read signature file.", mlog.String("path", signaturePath), mlog.Err(appErr))
+			return nil, nil, appErr
+		}
+		signatures = append(signatures, sigReader)
+	}
+
+	closeFunc = func() {
+		for _, signature := range signatures {
+			readCloseSeeker, ok := signature.(filesstore.ReadCloseSeeker)
+			if !ok {
+				mlog.Debug("Failed to cast signature to ReadCloseSeeker.", mlog.String("paths", fmt.Sprint(plugin.signaturePaths)), mlog.String("pluginId", plugin.pluginId))
+			}
+			readCloseSeeker.Close()
+		}
+	}
+	return signatures, closeFunc, nil
+}
+
+func (a *App) getPluginsFromFolder() (map[string]*pluginSignaturePaths, *model.AppError) {
+	fileStorePaths, appErr := a.ListDirectory(fileStorePluginFolder)
+	if appErr != nil {
+		return nil, model.NewAppError("getPluginsFromDir", "app.plugin.sync.list_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
+	}
+	pluginSignaturePathMap := make(map[string]*pluginSignaturePaths)
+	for _, path := range fileStorePaths {
+		if strings.HasSuffix(path, ".tar.gz") {
+			id := strings.TrimSuffix(filepath.Base(path), ".tar.gz")
+			helper := &pluginSignaturePaths{
+				pluginId:       id,
+				path:           path,
+				signaturePaths: []string{},
+			}
+			pluginSignaturePathMap[id] = helper
+		}
+	}
+	for _, path := range fileStorePaths {
+		if strings.HasSuffix(path, ".sig") {
+			// Parse plugin id from .sig files, that are stored using getSignatureStorePath
+			id := strings.TrimSuffix(filepath.Base(path), ".sig")
+			index := strings.LastIndex(id, ".")
+			id = id[:index]
+			if val, ok := pluginSignaturePathMap[id]; !ok {
+				mlog.Error("Unknown signature", mlog.String("path", path))
+			} else {
+				val.signaturePaths = append(val.signaturePaths, path)
+			}
+		}
+	}
+	return pluginSignaturePathMap, nil
+}
```

Comment:
some of it is not applicable anymore after the rework. since it's just the filename i say we can continue doing it inline, especially since this is the only spot we extra the id.